### DISCUSSION TITLE
Best way to get unique row id ?

### DISCUSSION THREAD

**Discussion by justin-tufte** [TanStack/table#2219] [General]
It looks like using row index as a unique key causes some issues when data is changed and the table body re-renders. Because of this I was looking for some sort of built in function or property maybe that the table generates to track rows. The other reason a unique id is helpful is to get rid of the React warning `Warning: Each child in a list should have a unique "key" prop.`

So my question is, is there built in functionality to handle this or should I be providing my own `getRowID` function to the table in order to generate unique keys? My other option seems to be to enforce all data sets to have a unique key property.

**Comment by LexQzim** [comment]
@justin-tufte did you find an answer for your question? 
I'm currently facing the same problem. I've got a big table and need to remember and update each cell. Therefore I tought about to give the cellid the id of this data. But I don't know how. I only can define how my columnid will look like but this is to general.

I appreciate any tipps and advices.

**Comment by qqjay2017** [comment]
` import { nanoid } from 'nanoid' `
